server/schedule/hot_cache.go

```diff
@@ -134,26 +137,32 @@ func (w *HotSpotCache) isNeedUpdateStatCache(region *core.RegionInfo, flowBytes
 
 	if oldItem != nil {
 		newItem.HotDegree = oldItem.HotDegree + 1
+		newItem.Stats = oldItem.Stats
 	}
 	if flowBytes >= hotRegionThreshold {
 		if oldItem == nil {
 			w.incMetrics("add_item", kind)
+			newItem.Stats = core.NewRollingStats(rollingWindowsSize)
 		}
-		return true, newItem
+		newItem.Stats.Add(float64(flowBytes))
+		needUpdate = true
+		return needUpdate, newItem
```

Comment:
Why introduce the `needUpdate`? I think original `regurn true, newItem` is simpler.